mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/scaladsl/MqttSource.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming.scaladsl
+
+import akka.actor.ActorSystem
+import akka.{Done, NotUsed}
+import akka.event.{Logging, LoggingAdapter}
+import akka.stream.OverflowStrategy
+import akka.stream.alpakka.mqtt.streaming.impl.Setup
+import akka.stream.alpakka.mqtt.streaming._
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, Source, SourceQueueWithComplete, Tcp}
+
+import scala.collection.immutable
+import scala.concurrent.{Future, Promise}
+
+object MqttSource {
+
+  def atMostOnce(settings: MqttSessionSettings,
+                 clientId: String,
+                 subscriptions: immutable.Seq[(String, ControlPacketFlags)]): Source[Publish, Future[Done]] =
+    Setup
+      .source { implicit materializer => implicit attributes =>
+        implicit val system: ActorSystem = materializer.system
+        implicit val logging: LoggingAdapter = Logging.getLogger(system, this)
+
+        val mqttClientSession = ActorMqttClientSession(settings)
```

Comment:
I think it is super important to keep the session parameterised to this function. Session life cycles endure beyond connections.

Reply:
To add to this: when sending a Publish with a QoS > 0, the client session keeps track of whether the ACKs have taken place and will retry until they have. This concern spans connections.

Reply:
Further thinking on this... if you're managing the connection within the convenience function, which you are, then you don't need to pass the session around after all. Nice one.

Reply:
Ok, I'll try that instead now.